project-base/tests/FrontendApiBundle/Functional/Order/GetOrderAsUnauthenticatedCustomerUserTest.php

```diff
@@ -0,0 +1,112 @@
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Order\OrderFacade
+     * @inject
+     */
+    private $orderFacade;
+
+    public function testGetOrder(): void
+    {
+        list($uuid, $urlHash, $expectedOrderData) = $this->getOrderDataProvider();
+
+        $graphQlType = 'order';
+        $response = $this->getResponseContentForQuery($this->getOrderQuery($uuid, $urlHash));
+        $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
+        $responseData = $this->getResponseDataForGraphQlType($response, $graphQlType);
+
+        $this->assertArrayHasKey('status', $responseData);
+        $this->assertSame($expectedOrderData['status'], $responseData['status']);
+
+        $this->assertArrayHasKey('totalPrice', $responseData);
+        $this->assertArrayHasKey('priceWithVat', $responseData['totalPrice']);
+        $this->assertSame($expectedOrderData['totalPriceWithVat'], $responseData['totalPrice']['priceWithVat']);
+    }
+
+    public function testGetOrderReturnsError(): void
+    {
+        list($uuid, $urlHash, $expectedErrorMessage) = $this->getIncorrectOrderDataProvider();
+
+        $response = $this->getResponseContentForQuery($this->getOrderQuery($uuid, $urlHash));
+        $this->assertResponseContainsArrayOfErrors($response);
+        $errors = $this->getErrorsFromResponse($response);
+
+        $this->assertArrayHasKey(0, $errors);
+        $this->assertArrayHasKey('message', $errors[0]);
+        $this->assertSame($expectedErrorMessage, $errors[0]['message']);
+    }
+
+    /**
+     * @return array
+     */
+    public function getOrderDataProvider(): array
+    {
+        $data = [];
+        $orderIds = [7, 8, 9];
+        foreach ($orderIds as $orderId) {
+            $order = $this->orderFacade->getById($orderId);
+            $data[] = [
+                $order->getUuid(),
+                $order->getUrlHash(),
+                [
+                    'status' => $order->getStatus()->getName(),
+                    'totalPriceWithVat' => $order->getTotalPriceWithVat()->getAmount(),
+                ],
+            ];
+        }
+        return $data;
+    }
+
+    /**
+     * @return array
+     */
+    public function getIncorrectOrderDataProvider(): array
+    {
+        $anyUuid = '5277304e-d68c-4742-86dd-53b91ac05730';
+        return [
+            [
+                $anyUuid,
+                null,
+                'You need to be logged in or provide argument \'urlHash\'.',
+            ],
+            [
+                $anyUuid,
+                'foo',
+                'Order with UUID \'' . $anyUuid . '\' not found.',
+            ],
+        ];
+    }
+
+    /**
+     * @param string $uuid
+     * @param string|null $urlHash
+     * @return string
+     */
+    private function getOrderQuery(string $uuid, ?string $urlHash = null): string
+    {
+        if ($urlHash !== null) {
+            $type = 'order (uuid:"' . $uuid . '", urlHash:"' . $urlHash . '")';
```

Comment:
Please rename `$type` to something more describing, like `$query` or something like that

Reply:
Maybe `$type` is not the best naming here but is much better than `$query` as it is only for one part of the query (the type and filter otpions) - maybe `$graphQlTypeWithFilters`?